astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -140,10 +141,10 @@ def clean_typing_generic_mro(sequences: list[list[ClassDef]]) -> None:
 
 
 def clean_duplicates_mro(
-    sequences: Iterable[Iterable[ClassDef]],
+    sequences: list[list[ClassDef]],
     cls: ClassDef,
     context: InferenceContext | None,
-) -> Iterable[Iterable[ClassDef]]:
+) -> list[list[ClassDef]]:
```

Comment:
Why would we narrow these?

Reply:
`mypy` was very unhappy about assigning the return value to something that was previously a list. Then I tried to fix that but something else broken. I didn't want to spend more time, and since it is used only once I think it doesn't matter too much.

Reply:
Np